src/omv/ports/stm32/sensor.c

```diff
@@ -1424,14 +1442,21 @@ int sensor_snapshot(sensor_t *sensor, image_t *image, uint32_t flags)
         __HAL_DCMI_ENABLE_IT(&DCMIHandle, DCMI_IT_FRAME);
     }
 
+    vbuffer_t *buffer;
+
     // Wait for the frame data. __WFI() below will exit right on time because of DCMI_IT_FRAME.
     // While waiting SysTick will trigger allowing us to timeout.
-    for (tick_start = HAL_GetTick(); waiting_for_data; ) {
+    for (tick_start = HAL_GetTick();;) {
+        buffer = framebuffer_get_head(FB_NO_FLAGS);
+
+        if (buffer) {
+            break;
+        }
+
```

Comment:
```suggestion
    for (tick_start = HAL_GetTick(); buffer == NULL;) {
        buffer = framebuffer_get_head(FB_NO_FLAGS);
```

Reply:
Wait, this does an extra __WFI().

Will fix with MDMA update.